Add instructions for compiling with ANGLE on Windows. (#8619)
godotengine/godot-docs: contributing/development/compiling/compiling_for_windows.rst

          :ref:`doc_data_paths_self_contained_mode` by creating a file called
          ``._sc_`` or ``_sc_`` in the ``bin/`` folder.

Compiling with ANGLE support
----------------------------

> ANGLE provides translation layer from OpenGL ES 3.x to Direct3D 11 and can be used

Reviewer: ```suggestion
ANGLE provides a translation layer from OpenGL ES 3.x to Direct3D 11 and can be used
```

> to provide better support for Compatibility renderer on some older GPUs with outdated

Reviewer: ```suggestion
to improve support for the Compatibility renderer on some older GPUs with outdated
```

OpenGL drivers and on Windows for ARM.

By default, Godot is built with dynamically linked ANGLE, you can use it by placing
``libEGL.dll`` and ``libGLESv2.dll`` alongside the executable.

.. note:: You can use dynamically linked ANGLE with export templates as well, rename
          aforementioned DLLs to ``libEGL.{architecture}.dll`` and ``libGLESv2.{architecture}.dll``
          and place them alongside export template executables, and libraries will
          be automatically copied during the export process.

To compile Godot with statically linked ANGLE:

> - Download pre-built static libraries from `godot-angle-static library <https://github.com/godotengine/godot-angle-static/releases>`_, and unzip it. 

Reviewer: ```suggestion
- Download pre-built static libraries from `godot-angle-static library <https://github.com/godotengine/godot-angle-static/releases>`_, and unzip them. 
```
